libs/storage/Tsavorite/cs/src/core/Index/Tsavorite/Implementation/BlockAllocate.cs

```diff
@@ -20,32 +20,28 @@ private static bool TryBlockAllocate<TInput, TOutput, TContext>(
                 out OperationStatus internalStatus)
         {
             pendingContext.flushEvent = allocator.FlushEvent;
-            logicalAddress = allocator.TryAllocate(recordSize);
+            logicalAddress = allocator.TryAllocateRetryNow(recordSize);
             if (logicalAddress > 0)
             {
                 pendingContext.flushEvent = default;
                 internalStatus = OperationStatus.SUCCESS;
                 return true;
             }
 
-            if (logicalAddress == 0)
-            {
-                // We expect flushEvent to be signaled.
-                internalStatus = OperationStatus.ALLOCATE_FAILED;
-                return false;
-            }
-
-            // logicalAddress is < 0 so we do not expect flushEvent to be signaled; return RETRY_LATER to refresh the epoch.
-            pendingContext.flushEvent = default;
-            allocator.TryComplete();
-            internalStatus = OperationStatus.RETRY_LATER;
+            Debug.Assert(logicalAddress == 0);
```

Comment:
Add a message to the Assert e.g. "<0 (RETRY_NOW) should already have been handled"